pkg/models/networks.go

```diff
@@ -77,38 +77,35 @@ type CreateNetwork struct {
 	Name                string               `json:"name" tf:"name"`
 	Description         string               `json:"description,omitempty" tf:"description"`
 	DisplayName         string               `json:"displayName,omitempty" tf:"display_name"`
-	CloudID             int                  `json:"-" tf:"cloud_id"`
-	GroupID             string               `json:"-" tf:"group_id"`
-	TypeID              int                  `json:"-" tf:"type_id"`
-	PoolID              int                  `json:"pool,omitempty" tf:"pool_id"`
-	NetworkDomainID     int                  `json:"-" tf:"domain_id"`
-	Site                IDStringModel        `json:"site,omitempty"`
-	Type                IDModel              `json:"type,omitempty"`
-	NetworkDomain       *IDModel             `json:"networkDomain,omitempty"`
-	NetworkProxy        *IDModel             `json:"networkProxy,omitempty"`
-	NetworkServer       IDModel              `json:"networkServer,omitempty"`
-	NetworkPool         PoolModel            `json:"networkPool,omitempty"`
-	NetworkProxyID      int                  `json:"-" tf:"proxy_id"`
-	ProxyID             int                  `json:"-" tf:"proxy_id"`
+	Site                IDStringModel        `json:"site" tf:"group"`
+	Type                IDModel              `json:"type" tf:"type"`
+	NetworkDomain       *IDModel             `json:"networkDomain" tf:"domain"`
+	NetworkProxy        *IDModel             `json:"networkProxy" tf:"network_proxy"`
+	NetworkServer       IDModel              `json:"networkServer" tf:"network_server"`
 	SearchDomains       string               `json:"searchDomains,omitempty" tf:"search_domains"`
-	Cidr                string               `json:"cidr,omitempty" tf:"cidr"`
+	Cidr                string               `json:"cidr" tf:"gateway_cidr"`
 	Gateway             string               `json:"gateway,omitempty" tf:"gateway"`
 	DNSPrimary          string               `json:"dnsPrimary,omitempty" tf:"primary_dns"`
 	DNSSecondary        string               `json:"dnsSecondary,omitempty" tf:"secondary_dns"`
-	Config              *CreateNetworkConfig `json:"config,omitempty" tf:"config,sub"`
+	Config              *CreateNetworkConfig `json:"config" tf:"config,sub"`
 	Active              bool                 `json:"active" tf:"active"`
```

Comment:
my only worry about changing the tf fields is that it could break the existing automation if any of the teams have used ? what is the need to change the tf fields which were already existing ? Please have a good explanation about this.

Reply:
Use can create both DHCP and Static Networks. Ideally he should be able to create one at a time. Please add necessary validations for the same.
https://turbo-fortnight-4dcdc26d.pages.github.io/#tag/Networks/operation/CreatesANetwork

Reply:
updated the proper model for both  DHCP and Static Networks.